libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1447,6 +1468,55 @@ string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFu
 		.render();
 	});
 }
+string YulUtilFunctions::readFromStorageReferenceType(Type const& _type)
+{
+	solUnimplementedAssert(_type.category() == Type::Category::Struct, "");
+
+	string functionName = "read_from_storage_reference_type_" + _type.identifier();
+
+	auto const& structType = dynamic_cast<StructType const&>(_type);
+	solUnimplementedAssert(structType.location() == DataLocation::Memory, "");
```

Comment:
```suggestion
	solAssert(structType.location() == DataLocation::Memory, "");
```
Calldata is read-only, so we can't read into it - it is already in storage, so reading to storage would be a no-op (for creating a copy in storage, we'd need to know which slot to copy to) - so this *has* to be memory, hasn't it?